src/components/common/NetworkSelector/index.tsx

```diff
@@ -3,47 +3,230 @@ import { useDarkMode } from '@/hooks/useDarkMode'
+      <Typography variant="overline" color="border.main">
+        Testnets
+      </Typography>
+    </Divider>
+  )
+}
+
+const UndeployedNetworks = ({
+  deployedChains,
+  chains,
+  safeAddress,
+}: {
+  deployedChains: string[]
+  chains: ChainInfo[]
+  safeAddress: string
+}) => {
+  const [open, setOpen] = useState(false)
+  const [replayOnChain, setReplayOnChain] = useState<ChainInfo>()
+  const addressBook = useAddressBook()
+  const safeName = addressBook[safeAddress]
+  const deployedChainInfos = useMemo(
+    () => chains.filter((chain) => deployedChains.includes(chain.chainId)),
+    [chains, deployedChains],
+  )
+  const [safeCreationData, safeCreationDataError] = useSafeCreationData(safeAddress, deployedChainInfos)
+
+  const availableNetworks = useReplayableNetworks(safeCreationData)
+
+  const [testNets, prodNets] = useMemo(
+    () =>
+      partition(
+        availableNetworks.filter((network) => !deployedChains.includes(network.chainId)),
+        (config) => config.isTestnet,
+      ),
+    [availableNetworks, deployedChains],
+  )
+
+  const onSelect = (chain: ChainInfo) => {
+    setReplayOnChain(chain)
+  }
+
+  const renderMenuItem = useCallback(
+    (chainId: string, isSelected: boolean) => {
+      const chain = chains.find((chain) => chain.chainId === chainId)
+      if (!chain) return null
+      return (
+        <>
+          <MenuItem
+            key={chainId}
+            value={chainId}
+            sx={{ '&:hover': { backgroundColor: 'inherit' } }}
+            onClick={() => onSelect(chain)}
+          >
+            <Box className={css.item}>
+              <ChainIndicator responsive={isSelected} chainId={chain.chainId} inline />
+              <PlusIcon className={css.plusIcon} />
+            </Box>
+          </MenuItem>
+        </>
+      )
+    },
+    [chains],
+  )
+
+  if (safeCreationDataError) {
+    return (
+      <Box p="0px 16px">
+        <Typography color="text.secondary" fontSize="14px">
+          Adding another network is not possible for this Safe
+        </Typography>
+      </Box>
+    )
+  }
+
+  return (
+    <>
+      <ListSubheader className={css.undeployedNetworksHeader}>
+        <ButtonBase className={css.listSubHeader} onClick={() => setOpen((prev) => !prev)}>
+          <Stack direction="row" spacing={1} alignItems="center">
+            <div>Show all networks</div>
+            <ExpandMoreIcon
+              fontSize="small"
+              sx={{
+                transform: open ? 'rotate(180deg)' : undefined,
+              }}
+            />
+          </Stack>
+        </ButtonBase>
+      </ListSubheader>
+      <Collapse in={open} timeout="auto" unmountOnExit>
+        {!safeCreationData ? (
+          <Box p="0px 16px">
+            <NetworkSkeleton />
+            <NetworkSkeleton />
+          </Box>
+        ) : (
+          <>
+            {prodNets.map((chain) => renderMenuItem(chain.chainId, false))}
```

Comment:
Can we change the `renderMenuItem` from a function to be a component? Otherwise every time the parent component re-renders this component will be re-rendered again